app/code/Magento/InventoryCatalog/Test/Integration/SetDataToLegacyStockStatusAtSourceItemsSaveTest.php

```diff
@@ -90,14 +90,60 @@ protected function setUp()
     }
 
     /**
-     * Tests that legacy stock status data will updates with legacy source item data.
+     * Tests that legacy stock status data will updates with legacy source item data while the indexer is "On Update".
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryCatalog/Test/_files/source_items_on_default_source.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryIndexer/Test/_files/reindex_inventory.php
+     */
+    public function testSetDataIndexedOnUpdate()
+    {
+        $productSku = 'SKU-1';
```

Comment:
Just keep the old test and create a new on which enables indexer and calls old test.